src/portable/raspberrypi/rp2040/dcd_rp2040.c

```diff
@@ -70,56 +84,81 @@ static struct hw_endpoint *hw_endpoint_get_by_addr(uint8_t ep_addr)
   return hw_endpoint_get_by_num(num, dir);
 }
 
-static void _hw_endpoint_alloc(struct hw_endpoint *ep, uint8_t transfer_type)
+static uint _hw_endpoint_data_size(struct hw_endpoint *ep)
 {
-  // size must be multiple of 64
+    // size must be multiple of 64
   uint size = tu_div_ceil(ep->wMaxPacketSize, 64) * 64u;
 
   // double buffered Bulk endpoint
-  if ( transfer_type == TUSB_XFER_BULK )
+  if (ep->transfer_type == TUSB_XFER_BULK)
   {
     size *= 2u;
   }
 
-  ep->hw_data_buf = next_buffer_ptr;
-  next_buffer_ptr += size;
+  return size;
+}
+
+static void _hw_endpoint_alloc(struct hw_endpoint *ep)
+{
+  // size must be multiple of 64
+  uint size = _hw_endpoint_data_size(ep);
+
+  uint block_num = size / 64;
```

Comment:
The reason i suggested using a uint64_t was to use bit operations rather than a bunch of loops; e.g. something like:
```
uint block_count = size / 64;
uint64_t block_mask = (1ull << block_count) - 1;
for (size_t start = 0; start < DPRAM_BLOCKS - block_num; start++) {
    if  (!(dpram_state & block_mask)) {
       start_block = start;
       dpram_state |= block_mask;
    }
    block_mask <<= 1;
}
```

Reply:
FYI Cortex-M0 is very inefficient with bitwise operations, in addition `uint64_t` is not a native data type which needs runtime support like ABI I64Shl to do math operations. 
So it's better to compare the assembly output before decide which method is better :)

Reply:
GCC deals with it inline, so it is a bit better (smaller in debug mode, but slightly larger in release mode)

The intention here was to save RAM (8 bytes vs 64), however I do agree that perhaps it is undesirable in general to have much of any increase in code size since this feature is something most be clearly don't need (their endpoint use is largely static)

I therefore suggest we `#ifdef` this new code